rest_framework_json_api/views.py

```diff
@@ -98,6 +100,54 @@ def get_queryset(self, *args, **kwargs):
         return qs
 
 
+class RelatedMixin(object):
+    """
+    This mixin handles all related entities, whose Serializers are declared in "related_serializers"
+    """
+    related_serializers = {}
```

Comment:
could we not use `included_serailizers` on the defined serializer instead?

Reply:
I could, but I think it will be better to keep it separate. Otherwise all related entities described here will be available via `include` by default, and there is no way to remove it from `include` list.

Reply:
Would it be not more consistent from an api caller when a relationship can be either fetch per link and per include? An api caller can then decide what he prefers. This way it would also be DRY.

Reply:
I'm not against of that, but how do I add `included` serializer, without adding related link ?
Let's say I want GET api/post/1/?include=author, but at the same time I want api/post/1/author/ return 404 ?

The only way I see is to use `included_serializers` if `related_serializer` is None, and use `related_serializer` otherwise. Also we need to declare related_serializer as None by default.

Reply:
That's true this wouldn't be possible anymore to have a include parameter allowed but not a accessible relationship.
I am more wondering whether there is really use case for this where this is needed?

In the end it is the same data returned in included and as the relationship and with `RelationMixin` permissions are also defined by parent resource which is the same with included.
Or do you have a specific use case in mind where this could be handy?

Reply:
Well, probably I have the case. Let's say we want client to "enforce" fetching related resource from the "related" url rather then just by including it within parent resource. My example is:

```python
{
    "data": {
        "type": "product",
        "id": "1",
        "attributes": {
            ...
        },
        "relationships": {
            "reviews": {
                "data": [
                    {
                        "type": "product-review",
                        "id": "1"
                    },
# A lot of object identifiers
                    {
                        "type": "product-review",
                        "id": "1000"
                    },
                ],
                "meta": {
                    "count": 1000
                },
                "links": {
                    "self": "api/product/1/relationships/reviews/",
                    "related": "api/product/1/reviews/"
                }
            },
```
Let's say I do not want to give client an ability to fetch such a huge response from api/product/1/ endpoint, because it is cached by product id for example. Of source we can use full_url (with get params like `include=reviews`) as cache key in the case, but who knows how do people use it.

So, what I would do here is:
Use included_serializers by default when related_serializer is None. And use related_serializers when it is not None. Such a way we give both opportunities and people can use `included_serializers` for both inclusion and related links. When they need different behavior - they use related_serializers dict.
Sounds like a plan ? :)

Reply:
Still not sure on the use case :). But yeah that's a plan let's do it like you suggested. Users might come up with use case when they use it I guess.